src/test/groovy/org/prebid/server/functional/tests/TimeoutSpec.groovy

```diff
@@ -329,12 +316,112 @@ class TimeoutSpec extends BaseSpec {
+
+        and: "Bid response should shutdown by timeout from stored request"
+        def errors = bidResponse.ext?.errors
+        assert errors[ErrorType.GENERIC]*.code == [1]
+        assert errors[ErrorType.GENERIC]*.message == ["Timeout has been exceeded"]
+    }
+
+    def "PBS should choose min timeout form config for bidder request when in request value lowest that in auction.biddertmax.min"() {
+        given: "PBS config with percent = 50"
+        def minBidderTmax = PBSUtils.getRandomNumber(50, MAX_TIMEOUT)
+        def prebidServerService = pbsServiceFactory.getService(["auction.biddertmax.min": minBidderTmax as String])
+
+        and: "Idle request PBS for stabilize timeout"
+        prebidServerService.sendAuctionRequest(BidRequest.defaultBidRequest)
+
+        and: "Default basic BidRequest with generic bidder"
+        def timeout = PBSUtils.getRandomNumber(0, 50)
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            tmax = timeout
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = prebidServerService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain min value from auction.biddertmax.min config"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.tmax == minBidderTmax as Long
+
+        and: "PBS response should contain tmax from request"
+        assert bidResponse?.ext?.tmaxrequest == timeout as Long
+    }
+
+    def "PBS should change timeout for bidder due to percent in auction.biddertmax.percent"() {
+        given: "PBS config with percent = 50"
+        def percent = PBSUtils.getRandomNumber(0, 98)
+        def prebidServerService = pbsServiceFactory.getService(["auction.biddertmax.percent": percent as String]
+                + PBS_CONFIG)
+
+        and: "Idle request PBS for stabilize timeout"
+        prebidServerService.sendAuctionRequest(BidRequest.defaultBidRequest)
+
+        and: "Default basic BidRequest with generic bidder"
+        def timeout = getRandomTimeout()
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            tmax = timeout
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = prebidServerService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain percent of request value"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert checkInternalProcess(bidderRequest.tmax, getPercentOfValue(percent,timeout))
+
+        and: "PBS response should contain tmax from request"
+        assert bidResponse?.ext?.tmaxrequest == timeout as Long
+    }
+
+    private static long getPercentOfValue(int percent, int timeout) {
+        (percent * timeout) / 100.0 as Long
+    }
+
+    private static boolean checkInternalProcess(Long bidderRequestTmax,  Long requestTimeout){
```

Comment:
1) `Long` -> `long`. Please use primitives for method arguments if you are not expecting nulls there.
2) Method name is strange. What is internal process?